src/main/webapp/script.js

```diff
@@ -602,39 +606,110 @@ function hideDiffs(cy, elems, deletedNodes, deletedEdges, addedNodes, addedEdges
+  let errorText = "";
+  const query = document.getElementById(type + '-search').value;
+  let result;
+  if (query !== "") {
+    result = searchFunction(cy, query);
+    if (result) {
+      highlightElements(cy, result);
+    } else {
+      errorText = type + " does not exist.";
+    }
+  }
+  document.getElementById(type + '-error').innerText = errorText;
+  return result;
+}
+
+/**
+ * Searches and highlights/zooms a graph for specified node
+ *
+ * @param cy the graph to search through
+ * @param query the name of the node to search for
+ * @returns the result of the search.
  */
 function searchNode(cy, query) {
-  // reset nodes to default color
-  cy.nodes().forEach(node => {
-    node.style('background-color', 'blue');
-    node.style('opacity', '1')
-  });
-  const target = findNodeInGraph(cy, query);
-  if (target) {
-    cy.nodes().forEach(node => node.style('opacity', '0.25'));
-    target.style('background-color', 'olive');
-    target.style('opacity', '1');
+  let target = cy.$id(query);
+  if (target.length != 0) {
     cy.fit(target, 50);
-    return true;
-  } else {
-    // fits all nodes on screen
-    cy.fit(cy.nodes(), 50);
-    return false;
+    return target;
   }
 }
 
 /**
- * Finds element in cy graph by id
+ * Searches and highlights/zooms a graph for specified token
+ *
+ * @param cy the graph to search through
+ * @param query the name of the token to search for
+ * @returns the result of the search.
  */
-function findNodeInGraph(cy, id) {
-  if (id.length != 0) {
-    const target = cy.$('#' + id);
-    if (target.length != 0) {
-      return target;
+function searchToken(cy, query) {
+  let target = cy.collection();
+  cy.nodes().forEach(node => {
+    // check tokens field exists since deleted nodes that are still
+    // shown don't have a tokens field
+    if (node.data().tokens && node.data().tokens.includes(query)) {
+      target = target.add(node);
     }
+  });
+  if (target.length > 0) {
+    return target;
   }
-  return null;
+}
+
+/**
+ * Highlights collection of nodes/edges
+ *
+ * @param cy the graph that contains nodes/edges
+ * @param target collection of nodes to highlight
+ */
+function highlightElements(cy, target) {
+  cy.nodes().forEach(node => node.toggleClass('non-highlighted', true));
+
+  // highlight desired nodes
+  target.forEach(node => {
+    node.style('border-width', '4px');
+    node.toggleClass('non-highlighted', false);
+  });
+  cy.fit(target[0], 50);
+  document.getElementById('num-selected').innerText = "Number of nodes selected: " + target.length;
+
+  // highlight adjacent edges
+  target.connectedEdges().forEach(edge => {
+    edge.style('line-style', 'dashed');
+    edge.style('z-index', '2');
+  });
+}
+
+/**
+ * Resets collection of nodes/edges to default state
+ *
+ * @param cy the graph that contains nodes/edges
+ */
+function resetElements(cy) {
+  // reset node borders and opacity
+  cy.nodes().forEach(node => {
+    node.style('border-width', '0px');
```

Comment:
Just want to note that in another branch @haoyxc uses border-width as well to highlight "queried" nodes. We should remember to integrate this there.